eng/scripts/Language-Settings.ps1

```diff
@@ -37,7 +37,7 @@ function Get-GoModuleVersionInfo($modPath)
 function Get-GoModuleProperties($goModPath)
 {
   $goModPath = $goModPath -replace "\\", "/"
-  if ($goModPath -match "(?<modPath>sdk/(?<serviceDir>(resourcemanager/)?([^/]+/)?(?<modName>[^/]+$)))")
+  if ($goModPath -match "(?<modPath>sdk/(?<serviceDir>(\w*/)?([^/]+/)?(?<modName>[^/]+$)))")
```

Comment:
Why this change? We intentionally only had resourcemanager because they were the only approved projects that were deeper in paths.

Reply:
@benbp helped out here. This regex does not match for the multi directory (ie sdk/data/aztables). It will match for sdk/azcore and sdk/azidentity, but not further. Is there a need to keep the resourcemanager portion?

Reply:
This should be matching sdk/data/aztables just fine, if it weren't then that pipeline would be completely broken. Is there some evidence that it isn't working for them?

Reply:
@weshaggard https://regex101.com/r/yE8yp3/1

Reply:
@weshaggard I agree that it's worth investigating why the pipeline is working because this regex does not appear to work based on our testing in that regex101 link.

Reply:
Generally the path passed here is a directory not the full path to go.mod, see https://github.com/Azure/azure-sdk-for-go/blob/main/eng/scripts/Language-Settings.ps1#L108 as an example. If we want to support the full path as well we could add `(/go.mod)?` at the end of the regex.

Reply:
I passed in the directory and this worked as expected. Thanks for clarifying @weshaggard